In [40]:
#read data using pandas
import pandas as pd

dataset = pd.read_csv('diabetes.csv')

dataset.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [41]:
#shape of data
dataset.shape

(768, 9)

In [42]:
#null values
dataset.isnull().sum()

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

In [43]:
#classify data into dependent and independent

x = dataset.iloc[:, :-1]
y = dataset.iloc[:, -1]

In [44]:
#now split data into training and testing
from sklearn.model_selection import train_test_split

x_train, x_test, y_train, y_test = train_test_split(x, y, random_state=0, test_size=0.3)

print(x_train.shape)
print(y_train.shape)
print(x_test.shape)
print(y_test.shape)

(537, 8)
(537,)
(231, 8)
(231,)


In [45]:
#import classifier and fit data

from sklearn.linear_model import SGDClassifier

sgd = SGDClassifier()

#fit training data
sgd.fit(x_train, y_train)

#predict data
pred = sgd.predict(x_test)

In [46]:
#calculate ccuracy
from sklearn.metrics import accuracy_score

accuracy_score(y_test, pred)

0.7186147186147186

we found the accuracy of the model to be 71.9%